vuepress/docs/next/docs/component-generator/blueprint-features.md

```diff
@@ -0,0 +1,135 @@
+
+## The Features:
+* Backend with Spring Boot 
+  * profiles (dev, prod)
+  * Pre-configured Cross-Origin Resource Sharing (CORS) settings
+  * Keycloak integration 
+  * Swagger/OpenAPI frontend
+  * Data modeling 
+  * Jhipster Design Language (JDL) support for schema upgrade
+  * Liquibase integration for schema upgrade
+* Frontend 
+  * Localization
+  * Keycloak integration
+  * .env profiles 
+
+For more information:
+* Install [Jhipster Entando Blueprint](https://github.com/entando/generator-jhipster-entando/blob/master/README.md)
+* Try implementing an Entando Blueprint-- [Create micro services and micro Frontents](../../tutorials/micro-frontends/react.html)
+
+## Premade Widgets
+When you create an entity using the Entando Blueprint, it generates a few
+premade widgets. Below, each will be reviewed.
+* [Authentication](#authentication)
+* [Custom events](#custom-events)
+* [Tests and mocks](#tests-and-mocks)
+* [PropTypes](#proptypes)
+* [Fetching data](#fetching-data)
+* [Form widget](#form-widget)
+
+We will be using `Conference` as an entity name for the example below.
+
+What the widgets have in **common**:
+* Each widget contains a README file that helps with the setup.
+
+* All generated widgets are web components created using Custom Elements
+API.
+
+* Each widget is displayed using the custom element tag - e.g., inside the
+details widget folder `conference/detailsWidget/public/index.html`, you
+can find
+`conference-details id="1" override-edit-handler hide-edit-button />`.
+This element `<conference-details />` is defined in the component entry
+point at
+`conference/detailsWidget/src/custom-elements/ConferenceDetailsElement.js`.
+
+> **Note**
+>
+> custom element names (`conference-details`) require a dash in them to
+> be used, (kebab-case) - they can not be single words.
+
+For more information about web components, custom elements and micro
+frontends, refer to [Create a React micro frontend widget](../../tutorials/micro-frontends/react.html)
+
+
+### Authentication
+
+If a widget requires authentication, the component is wrapped in
+`KeycloakContext.Provider` and the Keycloak object is fetched from the
+`window.entando.keycloak` variable. Entando uses Keycloak as the
+authentication provider, but you can add any provider as needed.
+
+    ReactDOM.render(
+      <KeycloakContext.Provider value={this.keycloak}>
+        <StylesProvider jss={this.jss}>
+          <ThemeProvider theme={this.muiTheme}>{FormContainer}</ThemeProvider>
+        </StylesProvider>
+      </KeycloakContext.Provider>,
+      this.mountPoint
+    );
+
+For more information about the authentication process, please refer
+to the [Authentication section](../reference/identity-management.html#authentication).
+
+### Custom events
+
+All widgets rely on custom events for communication which is why custom
```

Comment:
would reword... "All widgets rely on custom events for communication and contain event creation and removal, as well as event listener creation."